cmd/reviewdog/main.go

```diff
@@ -317,21 +317,57 @@ func githubService(ctx context.Context) (gs *githubservice.GitHubPullRequest, is
 	if err != nil {
 		return nil, isPR, err
 	}
-	// TODO: support commit build
-	if !isPR {
-		return nil, isPR, nil
-	}
 
 	client, err := githubClient(ctx, token)
 	if err != nil {
 		return nil, isPR, err
 	}
 
+	if !isPR {
+		prID, err := getPullRequestIDByBranchOrCommit(ctx, client, g)
+		if err != nil {
+			return nil, false, err
+		}
+		g.PullRequest = prID
+	}
+
 	gs, err = githubservice.NewGitHubPullRequest(client, g.Owner, g.Repo, g.PullRequest, g.SHA)
 	if err != nil {
-		return nil, isPR, err
+		return nil, false, err
+	}
+	return gs, true, nil
+}
+
+func getPullRequestIDByBranchOrCommit(ctx context.Context, client *github.Client, info *cienv.BuildInfo) (int, error) {
+	options := &github.PullRequestListOptions{
```

Comment:
I mean, specify PullRequestListOptions.Head and  PullRequestListOptions.Base.

Let GitHub API find p-r.

Reply:
Vexor doesn't provide information about the base branch name.
So, I can only use the head branch name to filter pull requests.

Reply:
How about using [the search api](https://help.github.com/en/articles/searching-issues-and-pull-requests#search-by-commit-sha) instead of the list pull request api?

The head of CI and the head of p-r may be different when CI task is queuing. The search API searches for pull requests that contain that SHA, not only the head.

for example, the following query returns this pull request.

```
curl 'https://api.github.com/search/issues?q=type:pr+repo:reviewdog/reviewdog+state:open+215db7e754ba43de540d7487018bbf20816490b1&sort=created&order=asc'
```

Reply:
Done.